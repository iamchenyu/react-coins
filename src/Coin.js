import React from "react";
import "./Coin.css";

const Coin = ({ coin }) =>
  coin ? (
    <img
      src={coin.src}
      alt={coin.alt}
      className="Coin"
      data-testid="coin-image"
    />
  ) : null;
export default Coin;
