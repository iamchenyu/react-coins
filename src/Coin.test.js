import React from "react";
import { render } from "@testing-library/react";
import Coin from "./Coin";

const head = {
  side: "head",
  src: "https://tinyurl.com/react-coin-heads-jpg",
  alt: "coin head image",
};

const tail = {
  side: "tail",
  src: "https://www.pcgs.com/UserImages/71009269r.jpg",
  alt: "coin tail image",
};

it("renders the component without crashing", () => {
  render(<Coin />);
  render(<Coin coin={head} />);
  render(<Coin coin={tail} />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<Coin />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot", () => {
  const { asFragment } = render(<Coin coin={head} />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot", () => {
  const { asFragment } = render(<Coin coin={tail} />);
  expect(asFragment()).toMatchSnapshot();
});
