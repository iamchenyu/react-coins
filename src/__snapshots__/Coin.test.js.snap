// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `<DocumentFragment />`;

exports[`matches the snapshot 2`] = `
<DocumentFragment>
  <img
    alt="coin head image"
    class="Coin"
    data-testid="coin-image"
    src="https://tinyurl.com/react-coin-heads-jpg"
  />
</DocumentFragment>
`;

exports[`matches the snapshot 3`] = `
<DocumentFragment>
  <img
    alt="coin tail image"
    class="Coin"
    data-testid="coin-image"
    src="https://www.pcgs.com/UserImages/71009269r.jpg"
  />
</DocumentFragment>
`;
